test-commons/src/main/java/com/sonarsource/checks/verifier/SingleFileVerifier.java

```diff
@@ -0,0 +1,157 @@
+   *
+   *   verifier.reportIssue("Issue on range with a secondary location").onRange(line, column, endLine, endColumn)
+   *     .addSecondary(secondary.line, secondary.column, secondary.endLine, secondary.endColumn, "Secondary message");
+   * </pre>
+   * @param message issue message
+   */
+  IssueBuilder reportIssue(String message);
+
+  /**
+   * Run the comparison and expect to find at least one issue.
+   */
+  void assertOneOrMoreIssues();
+
+  /**
+   * Run the comparison and expect to find no issue.
+   */
+  void assertNoIssues();
+
+  /**
+   * Must always call one and only one of: onFile, onLine, onRange
+   */
+  interface IssueBuilder {
+
+    /**
+     * issue is global to the source code file (not at a specific line number)
+     */
+    Issue onFile();
+
+    /**
+     * issue is related to a specific line number
+     * @param line, start at 1
+     */
+    Issue onLine(int line);
+
+    /**
+     * issue is at a precise issue location
+     * @param line, start at 1, line number of the first character
+     * @param column, start at 1, column number of the first character
+     * @param endLine, start at 1, line number of the last character, if the issue is on one line then endLine == line
+     * @param endColumn, start at 1, column number of the last character, if there's only one character then endColumn == column
+     */
+    Issue onRange(int line, int column, int endLine, int endColumn);
+  }
+
+  interface Issue {
+
+    /**
+     * @param gap Gap used for the computation of the effort (previously effortToFix)
+     */
+    Issue withGap(@Nullable Double gap);
+
+    /**
+     * Add a secondary location with an optional message
+     * @param line, start at 1, line number of the first character
+     * @param column, start at 1, column number of the first character
+     * @param endLine, start at 1, line number of the last character, if the location is on one line then endLine == line
+     * @param endColumn, start at 1, column number of the last character, if there's only one character then endColumn == column
+     * @param message optional message, can be null
+     */
+    Issue addSecondary(int line, int column, int endLine, int endColumn, @Nullable String message);
+
+    /**
+     * Add a flow location with an optional message
+     * @param line, start at 1, line number of the first character
+     * @param column, start at 1, column number of the first character
+     * @param endLine, start at 1, line number of the last character, if the location is on one line then endLine == line
+     * @param endColumn, start at 1, column number of the last character, if there's only one character then endColumn == column
+     * @param flowIndex, start at 1, to distinguish on which flow the location is appended
+     * @param message optional message, can be null
+     */
+    Issue addFlow(int line, int column, int endLine, int endColumn, int flowIndex, @Nullable String message);
```

Comment:
Does it work? If not, we should not expose this method.

Reply:
Removed.